### PR TITLE
Add storage cloning functionality

### PR DESCRIPTION
**Pull request by nvthongswansea**
- Add `gridscale_storage_clone` resource.
- Add `gridscale_storage_clone`'s acceptance tests.
- Add `gridscale_storage_clone`'s docs.

### CODE REVIEW

File: gridscale/resource_gridscale_storage_clone.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"source_storage_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				Description:  "ID of a storage instance which will be cloned",
```
Comment (bkircher) [inline gridscale/resource_gridscale_storage_clone.go]: Maybe rephrase to
```
"ID of the storage instance that will be cloned."
```
Also note trailing dot.

File: gridscale/resource_gridscale_storage_clone.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+			},
+			"capacity": {
+				Type:        schema.TypeInt,
+				Description: "The capacity of a storage in GB.",
+				Optional:    true,
+				Computed:    true,
+			},
+			"location_uuid": {
+				Type:        schema.TypeString,
+				Description: "Helps to identify which datacenter an object belongs to.",
```
Comment (bkircher) [inline gridscale/resource_gridscale_storage_clone.go]: Consider rephrasing to:

```
"Identifies the data center this object belongs to."
```

File: gridscale/resource_gridscale_storage_clone.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+					}
+					if !valid {
+						errors = append(errors, fmt.Errorf("%v is not a valid storage type. Valid types are: %v", v.(string), strings.Join(storageTypes, ",")))
+					}
+					return
+				},
+			},
+			"license_product_no": {
+				Type:        schema.TypeInt,
+				Description: "If a template has been used that requires a license key (e.g. Windows Servers) this shows the product_no of the license (see the /prices endpoint for more details).",
```
Comment (bkircher) [inline gridscale/resource_gridscale_storage_clone.go:72]: Maybe make `"product_no"` → `"product number"`

File: gridscale/resource_gridscale_storage_clone.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+	}
+}
+
+func resourceGridscaleStorageCloneCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gsclient.Client)
+	srcStorageID := d.Get("source_storage_id").(string)
+	ctx, cancel := context.WithTimeout(context.Background(), d.Timeout(schema.TimeoutCreate))
+	defer cancel()
```
Comment (bkircher) [inline gridscale/resource_gridscale_storage_clone.go:140]: :heart:

File: gridscale/resource_gridscale_storage_clone.go [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+	response, err := client.CloneStorage(ctx, srcStorageID)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.ObjectUUID)
+
+	log.Printf("A new storage %s has been created by cloning storage %s", response.ObjectUUID, srcStorageID)
+
+	// If the user wants a new name and new capcatity for the storage clone
```
Comment (bkircher) [inline gridscale/resource_gridscale_storage_clone.go]: capcatity → capacity

**Review by bkircher — APPROVED** [review]
Looks good to me. Just typos or wording. Please fix as you see fit and merge

File: gridscale/resource_gridscale_storage_clone.go (hunk 3 above)
Comment (nvthongswansea) [reply gridscale/resource_gridscale_storage_clone.go:72]: This would make it inconsistent with the `storage` resource (and the response from gridscale's server).
